apps/election-manager/src/lib/votecounting.ts

```diff
@@ -222,69 +222,80 @@ const buildVoteFromCvr = ({
 
 interface TallyParams {
   election: Election
-  precinctId?: string
-  scannerId?: string
   votes: VotesDict[]
+  filterContestsByParty?: string
 }
 
 export function tallyVotesByContest({
   election,
   votes,
+  filterContestsByParty,
 }: TallyParams): ContestTally[] {
   const contestTallies: ContestTally[] = []
   const { contests } = election
-  expandEitherNeitherContests(contests).forEach((contest) => {
-    let options: readonly ContestOption[] = []
-    if (contest.type === 'yesno') {
-      options = [['yes'], ['no']]
-    }
-    if (contest.type === 'candidate') {
-      options = contest.candidates
-    }
 
-    const tallies: ContestOptionTally[] = options
-      .map((option) => {
-        return { option, tally: 0 }
-      })
-      .concat(
-        contest.type === 'candidate' && contest.allowWriteIns
-          ? [{ option: writeInCandidate, tally: 0 }]
-          : []
-      )
+  const districtsForParty = election.ballotStyles
+    .filter((bs) => bs.partyId === filterContestsByParty)
+    .flatMap((bs) => bs.districts)
 
-    votes.forEach((vote) => {
-      const selected = vote[contest.id]
-      if (!selected) {
-        return
+  expandEitherNeitherContests(contests).forEach((contest) => {
+    if (
+      filterContestsByParty === undefined ||
+      (districtsForParty.includes(contest.districtId) &&
+        contest.partyId === filterContestsByParty)
+    ) {
+      let options: readonly ContestOption[] = []
```

Comment:
I'm not sure why github displaying these changes so weirdly but the code below this comment in this function is just whitespace changes